---
title: "INTEGRATION GSE 115469"
output: html_document
date: "2023-04-24"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}
library(ggplot2)
library(dplyr)
library(Seurat)
library(patchwork)
```

```{r GSE115469}
GSE115469_NF <- read.csv("C:/Users/sandr/Downloads/GSE115469_Data.csv", row.names = 1)

GSE115469_NF <- CreateSeuratObject(counts=GSE115469_NF, project="PEC2")
GSE115469_NF

GSE115469_NF <- subset(GSE115469_NF, subset = nFeature_RNA > 1)
View(GSE115469_NF@meta.data)


```

NO HAY GENES MT NI RB


```{r violinplot}
VlnPlot(GSE115469_NF, features = c("nFeature_RNA","nCount_RNA" ), ncol = 2)
```
Según las gráficas elegimos los puntos de corte para cada px
```{r}
plot1 <- FeatureScatter(GSE115469_NF, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot1
```

```{r subset}

P1TLH <- WhichCells(GSE115469_NF, expression = nFeature_RNA > 4000 & orig.ident == "P1TLH")
P2TLH <- WhichCells(GSE115469_NF, expression = nFeature_RNA > 1800 & orig.ident == "P2TLH")
P3TLH <- WhichCells(GSE115469_NF, expression = nFeature_RNA > 3000 & orig.ident == "P3TLH")
P4TLH <- WhichCells(GSE115469_NF, expression = nFeature_RNA > 3000 & orig.ident == "P4TLH")
P4TLHlow <- WhichCells(GSE115469_NF, expression = nFeature_RNA < 200 & orig.ident == "P4TLH")
P5TLH <- WhichCells(GSE115469_NF, expression = nFeature_RNA > 2000  & orig.ident == "P5TLH")
P5TLHlow <- WhichCells(GSE115469_NF, expression = nFeature_RNA < 200  & orig.ident == "P5TLH")

GSE115469 <- subset(GSE115469_NF, cells=setdiff(WhichCells(GSE115469_NF),c(P1TLH, P2TLH, P3TLH, P4TLH, P4TLHlow, P5TLH, P5TLHlow)))

ncol(GSE115469_NF)
ncol(GSE115469)

```
HEMOS PASADO DE 8444 A 8185 CELULAS

```{r }
VlnPlot(GSE115469, features = c("nFeature_RNA","nCount_RNA" ), ncol = 2)
plot1 <- FeatureScatter(GSE115469, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot1
```
```{r lista + }
GSE115469.list <- SplitObject(GSE115469, split.by = "orig.ident")


GSE115469.list <- lapply(X = GSE115469.list, FUN = function(x) {
    x <- NormalizeData(x)
    x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})


features <- SelectIntegrationFeatures(object.list = GSE115469.list)


```

```{r integracion}
immune.anchors <- FindIntegrationAnchors(object.list = GSE115469.list, anchor.features = features)

immune.combined <- IntegrateData(anchorset = immune.anchors)
```

```{r integrated analysis}

DefaultAssay(immune.combined) <- "integrated"

immune.combined <- ScaleData(immune.combined)
immune.combined <- RunPCA(immune.combined)

ElbowPlot(immune.combined)

immune.combined <- FindNeighbors(immune.combined, reduction = "pca", dims = 1:20)
immune.combined <- FindClusters(immune.combined, reduction = "pca")
immune.combined <- RunUMAP(immune.combined, reduction = "pca", dims = 1:20)

```
```{r visualizacion}
p1 <- DimPlot(immune.combined, reduction = "umap", group.by = "orig.ident")
p2 <- DimPlot(immune.combined, reduction = "umap", label = TRUE, repel = TRUE)
p1 + p2
```
```{r}
DimPlot(immune.combined, reduction = "umap", split.by = "orig.ident")
```
```{r ANNOTATION ScTypeload function}
# load gene set preparation function
source("https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R")
# load cell type annotation function
source("https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R")

```
```{r}
install.packages("xlsx", dependencies = TRUE)
library(xlsx)
```
```{r}
install.packages("openxlsx")
library("openxlsx")
```
```{r}
install.packages("HGNChelper")
library(HGNChelper)
```

```{r}
db_ = "https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx";
tissue = "Liver" 
# prepare gene sets
gs_list = gene_sets_prepare(db_, tissue)
```

```{r}
# get cell-type by cell matrix
es.max = sctype_score(scRNAseqData = immune.combined[["RNA"]]@scale.data, scaled = TRUE, 
                      gs = gs_list$gs_positive, gs2 = NULL) 

# NOTE: scRNAseqData parameter should correspond to your input scRNA-seq matrix. 
# In case Seurat is used, it is either pbmc[["RNA"]]@scale.data (default), pbmc[["SCT"]]@scale.data, in case sctransform is used for normalization,
# or pbmc[["integrated"]]@scale.data, in case a joint analysis of multiple single-cell datasets is performed.

# merge by cluster
cL_resutls = do.call("rbind", lapply(unique(immune.combined@meta.data$seurat_clusters), function(cl){
    es.max.cl = sort(rowSums(es.max[ ,rownames(immune.combined@meta.data[immune.combined@meta.data$seurat_clusters==cl, ])]), decreasing = !0)
    head(data.frame(cluster = cl, type = names(es.max.cl), scores = es.max.cl, ncells = sum(immune.combined@meta.data$seurat_clusters==cl)), 10)
}))
sctype_scores = cL_resutls %>% group_by(cluster) %>% top_n(n = 1, wt = scores)  

# set low-confident (low ScType score) clusters to "unknown"
sctype_scores$type[as.numeric(as.character(sctype_scores$scores)) < sctype_scores$ncells/4] = "Unknown"
print(sctype_scores[,1:3])
```












